sdk/search/search/src/searchIndexClient.ts

```diff
@@ -562,6 +578,50 @@ export class SearchIndexClient<T> {
     }
   }
 
+  private encodeContinuationToken(
+    nextLink: string | undefined,
+    nextPageParameters: SearchRequest | undefined
+  ): string | undefined {
+    if (!nextLink || !nextPageParameters) {
+      return undefined;
+    }
+    const payload = JSON.stringify({
+      apiVersion: this.apiVersion,
+      nextLink,
+      nextPageParameters
+    });
+    return encode(payload);
+  }
+
+  private decodeContinuationToken(
+    token?: string
+  ): { nextPageParameters: SearchRequest; nextLink: string } | undefined {
+    if (!token) {
+      return undefined;
+    }
+
+    const decodedToken = decode(token);
+
+    try {
+      const result: {
+        apiVersion: string;
+        nextLink: string;
+        nextPageParameters: SearchRequest;
+      } = JSON.parse(decodedToken);
+
+      if (result.apiVersion !== this.apiVersion) {
```

Comment:
It might be worth adding a maintenance note wherever apiVersion is defined to always update this code on every release.  We should talk on Thursday whether to hard code this to 2019-05-06-Preview (and later to just 2019-05-06) so it doesn't drift with versions when there's no algorithmic difference (i.e., if we use a token from next year's library with the code you have here even though nothing's changed in the "algorithm").

Reply:
I'll add a note next to the api version